### PR TITLE
#11085: Fix binary backward docs

### LINKED ISSUE #11085: Update documentation for Binary bw ops
Tracking Issue : #10585

### PR DESCRIPTION
**Pull request by Aswinmcw**
Fix binary backward docs
Tracking Issue : #10585 

### Checklist
- [ ] Post commit CI passes
- [ ] Model regression CI testing passes (if applicable)
- [ ] New/Existing tests provide coverage for changes

### CODE REVIEW

File: ttnn/cpp/ttnn/operations/eltwise/binary_backward/binary_backward_pybind.hpp [hunk 1]
```diff
@@ -398,7 +441,7 @@ void bind_binary_bw_mul(py::module& module, const binary_backward_operation_t& o
 }
 
 template <typename binary_backward_operation_t>
-void bind_binary_bw_div(py::module& module, const binary_backward_operation_t& operation, const std::string& description) {
+void bind_binary_bw_div(py::module& module, const binary_backward_operation_t& operation, const std::string& description, const std::string& supported_dtype) {
     auto doc = fmt::format(
```
Comment (eyonland) [inline ttnn/cpp/ttnn/operations/eltwise/binary_backward/binary_backward_pybind.hpp:445]: This looks like a candidate for a magic enum here.

File: ttnn/cpp/ttnn/operations/eltwise/binary_backward/binary_backward_pybind.hpp [hunk 2]
```diff
@@ -272,6 +301,12 @@ void bind_binary_backward_float_default(py::module& module, const binary_backwar
         Keyword args:
             * :attr:`memory_config` (Optional[ttnn.MemoryConfig]): memory config for the output tensor
 
+        Supported dtypes, layouts, and ranks:
+
+        {6}
+
+        Note : bfloat8_b/bfloat4_b supports only on TILE_LAYOUT
```
Comment (eyonland) [inline ttnn/cpp/ttnn/operations/eltwise/binary_backward/binary_backward_pybind.hpp:308]: is only supported on

File: ttnn/cpp/ttnn/operations/eltwise/binary_backward/binary_backward_pybind.hpp [hunk 3]
```diff
@@ -21,7 +21,7 @@ namespace binary_backward {
 namespace detail {
 
 template <typename binary_backward_operation_t>
-void bind_binary_backward_ops(py::module& module, const binary_backward_operation_t& operation, const std::string& description) {
+void bind_binary_backward_ops(py::module& module, const binary_backward_operation_t& operation, const std::string& description, const std::string& supported_dtype) {
```
Comment (patrickroberts) [inline ttnn/cpp/ttnn/operations/eltwise/binary_backward/binary_backward_pybind.hpp:24]: Please change all the
```cpp
const std::string& description, const std::string& supported_dtype
```
to
```cpp
std::string_view description, std::string_view supported_dtype
```

**Review by eyonland — APPROVED** [review]
Approving under assumption you follow recommendation from @patrickroberts

**Review by eyonland — APPROVED** [review]

**Pull request merged**